### ISSUE TITLE
So confused on how to set up ldm 

### ISSUE THREAD

**Issue by mindstormer12** [LemonBoy/ldm#56]
I apologize in advance if it is not appropriate to ask such a noob question here. Basically, I'm trying to set up automounting devices that I plug to my system and auto-dismounting them when I unplug it.

A question not related to ldm:

Does the latter even make sense? Would I have to manually dismount my devices no matter what if I want to unplug them? Is it necessary to dismount them if I'm not doing any writing on the drive (i.e. can I assume the system is not doing any background writing such as cache, and if so, does that mean I do not need to dismount)?

A question relating to ldm:

I cannot get ldm to work, despite getting using the provided service file (enabled and started it, I did not change its contents) and creating an /etc/ldm.conf with the appropriate GID and UID values. Is it possible someone can provide a straightforward guide on how to set it up once I install ldm? I am going to uninstall and delete every thing so I can start from scratch without things conflicting, since I guess I messed up somewhere.

Thanks.

**Comment by LemonBoy** [comment]
You should always unmount your devices before unplugging them, some filesystems don't like being left in a 'dirty' state.
And if you're using the latest git version there's no need to specify the gid/uid anymore, just replace those two entries in ldm.conf with one named `MOUNT_OWNER=<your user name>`.
(I actually forgot to update the readme)

**Comment by mindstormer12** [comment]
I'm using the stable version from the AUR (https://aur.archlinux.org/packages/ldm/, I prefer to use this one). My service file is literally the one copied from github (I didn't change it at all) and my /etc/ldm.conf is:

USER_GID = 1000
USER_UID = 1000
BASE_MOUNTPOINT = /mnt

I did enabled the ldm.service and tried to start it but failed--systemctl status ldm.service says I must provide gid and uid. I can start ldm in a given session by ldm -d -g 1000 -u 1000 but cannot get ldm to autostart as a service.

**Comment by LemonBoy** [comment]
Can you please paste your service file and the conf file ?

**Comment by mindstormer12** [comment]
I did not edit the service file, so my /etc/systemd/system/ldm.service is:

>[Unit]
>Description=lightweight device mounter
>
>[Service]
>EnvironmentFile=/etc/ldm.conf
>ExecStart=/usr/bin/ldm -u ${MOUNT_OWNER} -p ${BASE_MOUNTPOINT}
>KillMode=process
>
>[Install]
>WantedBy=multi-user.target

My /etc/ldm.conf is: 

>USER_GID = 1000
>USER_UID = 1000
>BASE_MOUNTPOINT = /mnt

I enabled the service but when I started it via `systemctl start ldm.service`, I get an error. `systemctl status ldm.service` says I must specify GID and UID. I shouldn't need to use `ldm -d -g 1000 -u 1000`, which only works for the current session, if I can get the service to work, right?

**Comment by LemonBoy** [comment]
Well, this is self explaining, the service refers to `MOUNT_OWNER` and your conf file is still in the GID/UID format, you're using the newer .service with an older version of ldm.
The package should come with it's own service file, if it doesn't please tell me.

**Comment by mindstormer12** [comment]
I think I understand now--I used the ldm package of the latest version from the AUR (are you TheLemonMan--the maintainer for it?) but couldn't find the service file. I'm guessing the the service file is only provided if I `git clone` the ldm repo and compile via `make install` then, right? Therefore, I go to the github page and used the service file provided but that was changed since the last official version so now I am using the older service file. If there was a ldm-git for the AUR, then every commit you make will mean there's an update available for the user? I'm new to Linux (I apologize) so I hope you can confirm my understanding and tell me whether I'm approaching the issue correctly as an Arch user (I saw your thread on the Arch Linux forum so I assume you are one too).

Automounting works as expected now, but I have problems unmounting. When I do `umount /mnt/usb`, I get `umount failed: Operation not permitted`. When I do the same command with `sudo`, the command goes through and returns no errors, but the device is not unmounted. I still see the directory and the files of the device.

I appreciate your patience.

**Comment by LemonBoy** [comment]
I'll have a look at the package since it should come with a systemd service.
A -git package means it'll always pull the most recent revision every time you makepkg it,, the version itself doesn't get updated after each commit but rather done manually from the package maintainer, usually after each point release. I used to maintain a git package too but decided not to bother and to release every few months as ldm is pretty stable. 
Please have a look at the ldmc man page to find out how to properly unmount a device.

**Comment by mindstormer12** [comment]
Hmm, `ldmc -r /mnt/usb` doesn't seem to work--it says "The operation didn't complete successfully".

**Comment by LemonBoy** [comment]
The service is installed in the default folder, you can find it at `usr/lib/systemd/system/ldm.service`.
If the device shows up in `ldmc -l` and `lsof /mnt/usb` shows nothing then it's a bug, otherwise there's just another application that's using the disk and thus preventing ldm to do its work.

**Comment by mindstormer12** [comment]
Really weird... I did `ldm -l` as you've said and it didn't show the device. I then did `sudo umount -f /mnt/usb` which didn't return an error but device is still mounted. I did `ldm -l` again and now it shows the device, so then I tried `ldm -r /mnt/usb`, which says "Operation completed successfully" but it is still mounted (`ldm -l` after still showed the device). 

This is all on my laptop and in the console, so I don't know what application could possibly be preventing ldm from doing its work. If `sudo umount -f /mnt/usb` is not working either, then maybe I should just try something like udiskie?
